File: spotwo.java
/*This code is written by @Author Shivam Sharma(28shivamsharma@gmail.com).*
* This code has proprietary right to Author.*
* Read LICENCE to use it.*/
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.Arrays;
import java.math.BigInteger;
class Main
{
	static BigInteger xpown(BigInteger x,BigInteger n)
	{
		BigInteger a;
		if(n.equals(new BigInteger("0")))
		{
			return new BigInteger("1");
		}
		if(n.equals(new BigInteger("1")))
		{
			return x;
		}
		a=xpown(x,n.divide(new BigInteger("2")));
		a=(a.multiply(a)).mod(new BigInteger("1000000007"));
		if(n.mod(new BigInteger("2")).equals(new BigInteger("1")))
		{
			a=(a.multiply(x)).mod(new BigInteger("1000000007"));
		}
		return a;
	}
	public static void main(String[] args)throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		int tc=Integer.parseInt(br.readLine());
		BigInteger bi;
		while(tc-->0)
		{
			int n=Integer.parseInt(br.readLine());
			bi=new BigInteger(Integer.toBinaryString(n));
			bi=bi.multiply(new BigInteger("2"));
			System.out.println(xpown(new BigInteger("2"),bi).toString());
		}
	}
}

File: spotwo.c
/*This code is written by @Author Shivam Sharma(28shivamsharma@gmail.com).*
* This code has proprietary right to Author.*
* Read LICENCE to use it.*/
#include <stdio.h>
#define mod 1000000007
#define ull unsigned long long int
int print_binary(int n,int *p)
{
	int i,l,len=0,temp=n;
	while(temp/2>0)
	{
		len++;
		temp/=2;
	}
	len++;
	l=len;
	while(len>0)
	{
		p[len-1]=n%2;
		n/=2;
		len--;
	}
	return l;
}
ull binary_no(int n)
{
	int p[26],j,len;
	ull bi=0;
	len=print_binary(n,p);
	for(j=0;j<len;j++)
	{
		bi+=(ull)p[j]*(ull)pow(10,len-j-1);
	}
	return bi;
}

ull xpown(ull x,ull n)
{
	ull a;
	if(n==0)
	{
		return 1;
	}
	if(n==1)
	{
		return x;
	}
	a=xpown(x,n/2);
	a=(a*a)%mod;
	if(n%2==1)
	{
		a=(a*x)%mod;
	}
	return a;
}
int main()
{
	int tc;
	scanf("%d",&tc);
	while(tc--)
	{
		int n;
		scanf("%d",&n);
		ull temp=(xpown(2,binary_no(n))%mod*xpown(2,binary_no(n))%mod)%mod;
		printf("%llu\n",temp);
	}
	return 0;
}